sdks/router-sdk/src/entities/trade.ts

```diff
@@ -155,13 +155,49 @@ export class Trade<TInput extends Currency, TOutput extends Currency, TTradeType
 
     const outputCurrency = this.swaps[0].outputAmount.currency
     const totalOutputFromRoutes = this.swaps
-      .map(({ outputAmount }) => outputAmount)
+      .map(({ outputAmount: routeOutputAmount }) => routeOutputAmount)
       .reduce((total, cur) => total.add(cur), CurrencyAmount.fromRawAmount(outputCurrency, 0))
 
     this._outputAmount = totalOutputFromRoutes
     return this._outputAmount
   }
 
+  /**
+   * Returns the sum of all swaps within the trade where the pathInput/Output is native
+   * @returns
+   * inputAmount: total input amount
+   * inputAmountNative: total amount of native currency required for ETH input paths
```

Comment:
so these should only ever be nonzero when the route is v4, yes?

Reply:
or sorry i guess im confusing this with pathInput/pathOutput... 

these actually return the route input/output amounts so native could be non zero for v2/v3, and then require a wrap/unwrap.. yes?

Reply:
native should always be 0 for v2/v3